### ISSUE TITLE
Questions Regarding the oepnvla and openvla-oft Papers

### ISSUE THREAD

**Issue by MstarLioning** [moojink/openvla-oft#17]
Hello, thank you for writing the two papers, oepnvla and openvla-oft. I have a series of questions regarding these two papers:

1、In openVLA, according to this issue (https://github.com/openvla/openvla/issues/225), if the natural language description of a task is slightly modified, all tasks still execute according to the training text. Meanwhile, if the natural language description is altered more significantly, the robotic arm exhibits erratic shaking (and never executes the task correctly). How can this be resolved? I have also observed a similar issue in this discussion (https://github.com/openvla/openvla/issues/197). Could you please provide some guidance to help me understand what mistake I might be making?

2、In the application of openVLA within LIBERO, why is the uploaded image "agentview_image" used instead of "robot0_eye"? Wouldn't using "robot0_eye" better simulate the real scenario?

3、In oepnvla-OFT, I noticed that you mentioned results from running π0 on LIBERO. Is there any explicit procedure available to reproduce these results?

I look forward to your reply and am extremely grateful for your help!

**Comment by MstarLioning** [comment]
Additionally,I noticed that in OFT-OpenVLA you mentioned that the predictions for actions are output as continuous values directly. Given this, why is there still an unnorm_key? In the original version, wasn’t the role of unnorm_key to provide statistical information to map the output buckets to a fixed interval? I look forward to your reply! Thank you!

**Comment by LukeLIN-web** [comment]
> Additionally,I noticed that in OFT-OpenVLA you mentioned that the predictions for actions are output as continuous values directly. Given this, why is there still an unnorm_key? In the original version, wasn’t the role of unnorm_key to provide statistical information to map the output buckets to a fixed interval? I look forward to your reply! Thank you!

unnorm key is to unorm the value according to different task. Althought I don't understand why we need unnorm.

**Comment by moojink** [comment]
Hi @MstarLioning,

1. If you change the language instruction that is fed into the policy, how well the policy does depends on whether it can generalize to the new instruction, which depends on the breadth of instructions seen in the training dataset. For OpenVLA, generalizing to new instructions works well for certain robots and environments (like Bridge WidowX and Google RT-X) because the corresponding datasets covered many different tasks. For OpenVLA-OFT (or really any policy that is fine-tuned on a narrow target dataset), generalization to new language instructions may be more limited, as expected since each policy is fine-tuned separately on a small set of demos in the target domain.
2. For OpenVLA-OFT, we actually use both `"agentview_image"` (third-person) and `"robot0_eye_in_hand_image"` (wrist camera image). We only use `"agentview_image"` for OpenVLA because it doesn't by default support multi-image inputs.
3. Please see the $\pi_0$ repo for instructions on reproducing their LIBERO results: https://github.com/Physical-Intelligence/openpi/tree/main/examples/libero

> Additionally,I noticed that in OFT-OpenVLA you mentioned that the predictions for actions are output as continuous values directly. Given this, why is there still an unnorm_key? In the original version, wasn’t the role of unnorm_key to provide statistical information to map the output buckets to a fixed interval? I look forward to your reply! Thank you!

We normalize actions to the range [-1,+1] at train time regardless of the action representation (discrete vs. continuous). OpenVLA and OpenVLA-OFT both do this. Because of this, we need the norm stats to un-normalize at test time to bring actions back to their proper range.
